src/disco/shred/fd_shred_tile.c

```diff
@@ -0,0 +1,694 @@
+  }
+}
+
+#define NET_IN_IDX 0
+#define MAX_SLOTS_PER_EPOCH 432000UL
+
+static void
+before_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq,
+             ulong  sig,
+             int *  opt_filter ) {
+  (void)seq;
+
+  shred_tile_ctx_t * ctx = (shred_tile_ctx_t *)_ctx;
+
+  *opt_filter = (in_idx==NET_IN_IDX) & (fd_disco_netmux_sig_port( sig )!=ctx->args->shred_listen_port);
+}
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+  (void)sig;
+  (void)opt_filter;
+
+  shred_tile_ctx_t * ctx = (shred_tile_ctx_t *)_ctx;
+
+  if( FD_UNLIKELY( in_idx!=NET_IN_IDX ) ) {
+    /* We can just go ahead and shred it here, even though we may get
+       overrun.  If we do end up getting overrun, we just won't send
+       these shreds out and we'll reuse the FEC set for the next one.
+       From a higher level though, if we do get overrun, a bunch of
+       shreds will never be transmitted, and we'll end up producing a
+       block that never lands on chain. */
+    fd_fec_set_t * out = ctx->fec_sets + ctx->shredder_fec_set_idx;
+
+    uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->bank_shred_wksp, chunk );
+    fd_entry_batch_meta_t const * entry_batch_meta = (fd_entry_batch_meta_t const *)dcache_entry;
+    uchar const *                 entry_batch      = dcache_entry + sizeof(fd_entry_batch_meta_t);
+    ulong                         entry_batch_sz   = sz           - sizeof(fd_entry_batch_meta_t);
+
+    FD_LOG_INFO(( "Shredding batch (%lu, %lu, %lu, %lu, %lu)", entry_batch_meta->slot, entry_batch_meta->parent_offset, entry_batch_meta->bank_max_tick_height, entry_batch_meta->reference_tick, entry_batch_meta->tick ));
+
+    fd_shredder_init_batch( ctx->shredder, entry_batch, entry_batch_sz, entry_batch_meta );
+
+    /* We're depending on the pack tile to produce microblocks that can
+       fit in 1 FEC set. */
+    FD_TEST( fd_shredder_next_fec_set( ctx->shredder, ctx->args->shred_signing_key, out ) );
+    fd_shredder_fini_batch( ctx->shredder );
+
+    d_rcvd_join( d_rcvd_new( d_rcvd_delete( d_rcvd_leave( out->data_shred_rcvd   ) ) ) );
+    p_rcvd_join( p_rcvd_new( p_rcvd_delete( p_rcvd_leave( out->parity_shred_rcvd ) ) ) );
+
+    ctx->send_fec_set_idx = ctx->shredder_fec_set_idx;
+
+    ctx->last_frag_was_shred = 0;
+  } else {
+    /* The FEC resolver API does not present a prepare/commit model. If we
+       get overrun between when the FEC resolver verifies the signature
+       and when it stores the local copy, we could end up storing and
+       retransmitting garbage.  Instead we copy it locally, sadly, and
+       only give it to the FEC resolver when we know it won't be overrun
+       anymore. */
+    uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->net_wksp, chunk );
```

Comment:
Can you validate that chunk is in bounds of the in wksp here? You can look at eg, dedup or verify to see how other tiles do this.